app/services/checksum_validator.rb

```diff
@@ -140,4 +145,30 @@ def latest_signature_catalog_entries
   def latest_moab_version
     moab_storage_object.version_list.last
   end
+
+  def validate_against_signature_catalog(data_content_file)
+    absent_from_manifest_data = { file_path: data_content_file, manifest_file_path: latest_signature_catalog_path }
+    paths_from_manifest = latest_signature_catalog_entries.map { |entry| signature_catalog_entry_path(entry) }
```

Comment:
you call this method for each file;  it seems like you could have a method to memoize the paths from signature catalog. (and I would call it paths_from_signature_catalog)